lib/core/macros.nim

```diff
@@ -1435,9 +1435,35 @@ proc hasArgOfName*(params: NimNode; name: string): bool {.compileTime.}=
   ## Search ``nnkFormalParams`` for an argument.
   expectKind(params, nnkFormalParams)
   for i in 1 ..< params.len:
-    template node: untyped = params[i]
-    if name.eqIdent( $ node[0]):
-      return true
+    # In, for example, proc p(s:string;i,j,k:int) ..., the i,j and k arguments
```

Comment:
I'd remove this whole comment, nothing surprising is going on here (and other more complicated procs don't do that either)